### DISCUSSION TITLE
Error handling

### DISCUSSION THREAD

**Discussion by josejaviersanahuja** [freshframework/fresh#1466] [General]
Dear all.

Thank you in advance.

I want to understand how to catch errors and render a custom message. I have read the docs, and is very short. I created the file routes/_500.tsx with a default export and ErrorPageProps etc etc but even when there were many errors during development, it never rendered.

So... I have tried throwing error inside a Page Handler inside a useEffect, inside an onClick  function.

What do i have to do to throw an error that will trigger my custom error  page?

Thanks in advance

**Comment by marvinhagemeister** [comment]
> What do i have to do to throw an error that will trigger my custom error page?

Wait for Fresh 1.3 to be released (likely this week). There we improved error handling in that errors thrown from handlers will always render the `_500.tsx` template.
